test/tests/functional/pbs_cray_smoketest.py

```diff
@@ -0,0 +1,132 @@
+        nodes_up_apstat = cu.count_node_in_state('up')
+        nodes = self.server.filter(NODE,
+                                   {ATTR_rescavail+'.vntype': 'cray_compute'})
+        self.assertEqual(nodes_up_apstat,
+                         len(nodes[ATTR_rescavail+'.vntype=cray_compute']))
+
+    @tags('cray', 'smoke')
+    def test_cray_pbsnodes(self):
+        """
+        Check pbsnodes -av output on a Cray system.
+        pbsnodes output should show that nodes are free and resources are
+        available.
+        """
+        nodes = self.server.status(NODE)
+        for node in nodes:
+            self.assertEqual(node['state'], 'free')
+            self.assertEqual(node['resources_assigned.ncpus'], '0')
+            self.assertEqual(node['resources_assigned.mem'], '0kb')
+
+    @tags('cray', 'smoke')
+    def test_cray_login_job(self):
+        """
+        Submit a simple sleep job that requests to run on a login node
+        and expect that job to go in running state on a login node.
+        """
+        j1 = Job(TEST_USER, {ATTR_l + '.vntype': 'cray_login'})
+        jid1 = self.server.submit(j1)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid1)
+        # fetch node name where the job is running and check that the
+        # node is a login node
+        qstat = self.server.status(JOB, 'exec_vnode', id=jid1)
+        vname = qstat[0]['exec_vnode'].partition(':')[0].strip('(')
+        self.server.expect(NODE, {ATTR_rescavail + '.vntype': 'cray_login'},
+                           id=vname)
+
+        cu = CrayUtils()
+        # Check if number of compute nodes in use are 0
+        self.assertEqual(cu.count_node_in_state('use'), 0)
+
+    @tags('cray', 'smoke')
+    def test_cray_compute_job(self):
+        """
+        Submit a simple sleep job that runs on a compute node and
+        expect the job to go in running state on a compute node.
+        """
+        j1 = Job(TEST_USER, {ATTR_l + '.vntype': 'cray_compute'})
+        jid1 = self.server.submit(j1)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid1)
+        # fetch node name where the job is running and check that the
+        # node is a compute node
+        qstat = self.server.status(JOB, 'exec_vnode', id=jid1)
+        vname = qstat[0]['exec_vnode'].partition(':')[0].strip('(')
+        self.server.expect(NODE, {ATTR_rescavail + '.vntype': 'cray_compute'},
+                           id=vname)
+        # Sleep for some time before aprun actually starts
+        # using the reservation
+        time.sleep(5)
+
+        cu = CrayUtils()
+        # Check if number of compute nodes in use are 1
+        self.assertEqual(cu.count_node_in_state('resv'), 1)
+        if self.du.get_platform() == 'cray':
+            # Cray simulator will not show anything in 'use' because
+            # aprun command is just a pass through on simulator
+            self.assertEqual(cu.count_node_in_state('use'), 1)
```

Comment:
Please add a check that job ends normally and job output and error files are saved correctly. After the job ends apstat should show there are no compute nodes in use.

Reply:
I'll check that the job ends and also verify output/error files generated by the job